storage/innobase/xtrabackup/src/redo_log.cc

```diff
@@ -325,17 +328,25 @@ ssize_t Redo_Log_Reader::read_logfile(bool is_last, bool *finished) {
     const lsn_t end_lsn =
         read_log_seg(log, log_buf + len, start_lsn, start_lsn + RECV_SCAN_SIZE);
 
+    if (end_lsn == 0) {
+      xb::error() << "read_log_seg() failed.";
+      m_error = true;
+      return (-1);
```

Comment:
what is the difference between return 0 and -1 ? Where do we deal with return code?

Reply:
so 0 means. We continue the operation -1 means some error.
same is mention in comment of `read_log`.
 @return read length or -1 if error. */